src/Validators/IntegrityChecks/Form/EmailConfigurationCheck.cs

```diff
@@ -0,0 +1,55 @@
+﻿using form_builder.Configuration;
+using form_builder.Enum;
+using form_builder.Models;
+using form_builder.Providers.Transforms.EmailConfiguration;
+
+namespace form_builder.Validators.IntegrityChecks.Form
+{
+    public class EmailConfigurationCheck : IFormSchemaIntegrityCheck
+    {
+        private readonly IEmailConfigurationTransformDataProvider _emailConfigProvider;
+
+        public EmailConfigurationCheck(IEmailConfigurationTransformDataProvider emailConfigProvider) =>
+            _emailConfigProvider = emailConfigProvider;
+
+        public IntegrityCheckResult Validate(FormSchema schema) => ValidateAsync(schema).Result;
+
+        public async Task<IntegrityCheckResult> ValidateAsync(FormSchema schema)
+        {
+            IntegrityCheckResult result = new();
+
+            bool containsEmail = schema.Pages
+                .Where(page => page.Behaviours is not null)
+                .SelectMany(page => page.Behaviours)
+                .Any(page => page.BehaviourType.Equals(EBehaviourType.SubmitAndEmail));
+
+            if (!containsEmail)
+                return result;
+
+            List<EmailConfiguration> emailConfig = await _emailConfigProvider.Get<List<EmailConfiguration>>();
+            EmailConfiguration formEmailConfig = emailConfig
+                .FirstOrDefault(email => email.FormName
+                    .Any(_ => _.Equals(schema.BaseURL.ToLower())));
```

Comment:
We've mostly swapped it out for passing a string comparator as an overload to the .Equals() method. 
eg. .Equals(schema.BaseURL, StringComparison.OrdinalIgnoreCase)